pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/RexExpression.java

```diff
@@ -49,6 +50,10 @@ static RexExpression toRexExpression(RexNode rexNode) {
     } else if (rexNode instanceof RexLiteral) {
       RexLiteral rexLiteral = ((RexLiteral) rexNode);
       FieldSpec.DataType dataType = RelToStageConverter.convertToFieldSpecDataType(rexLiteral.getType());
+      //Added to set the DataType as UNKNOWN
```

Comment:
@61yao, After setting the `UNKNOWN` datatype, the value gets set as String `null`., which later fails for the Numeric data type. 
I think you have made some changes in a couple of your PR. Do you think, once your PR will be merged this works fine?

Reply:
This is v2 engine code. @walterddr should have a better understanding of this. My code only deals with v1 path.

Reply:
@walterddr do you think your [PR](https://github.com/apache/pinot/pull/10423) is required here to merge before your change? Right now, it's failing because null is getting set as 'null' literal.